File: SYP.Web/Modules/ServerTypes/Products/ProductCategoryEditorColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProductCategoryRow } from "./ProductCategoryRow";

export interface ProductCategoryEditorColumns {
    Id: Column<ProductCategoryRow>;
    ParentName: Column<ProductCategoryRow>;
    Name: Column<ProductCategoryRow>;
    Path: Column<ProductCategoryRow>;
    FullPath: Column<ProductCategoryRow>;
    SortOrder: Column<ProductCategoryRow>;
    IsActive: Column<ProductCategoryRow>;
    CreatedAt: Column<ProductCategoryRow>;
}

export class ProductCategoryEditorColumns extends ColumnsBase<ProductCategoryRow> {
    static readonly columnsKey = 'Products.ProductCategoryEditor';
    static readonly Fields = fieldsProxy<ProductCategoryEditorColumns>();
}

File: SYP.Web/Modules/Catalog/ProductCategory/Editor/ProductCategoryEditorColumns.cs
namespace SYP.Catalog.Editor;

[ColumnsScript("Catalog.ProductCategoryEditor")]
[BasedOnRow(typeof(ProductCategoryRow), CheckNames = true)]
public class ProductCategoryEditorColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string ParentName { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string FullPath { get; set; }
    public int SortOrder { get; set; }
    public bool IsActive { get; set; }
    public DateTime CreatedAt { get; set; }
}